apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,438 @@
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-complete',
+      (transferInfo) => {
+        this._setProfileConnected(this.Profiles.OPP, false);
+        window.dispatchEvent(new CustomEvent('bluetooth-opp-transfer-complete',
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    // decouple bluetooth enable/disable function from other system part
+    window.addEventListener('request-enable-bluetooth',
+      this._enableHandler.bind(this));
+    window.addEventListener('request-disable-bluetooth',
+      this._disableHandler.bind(this));
+
+    Service.registerState('isEnabled', this);
+    // LazyLoader.load(['js/bluetooth_icon.js',
+    //                  'js/bluetooth_transfer_icon.js',
+    //                  'js/bluetooth_headphone_icon.js'], function() {
+    //   this.icon = new BluetoothIcon(this);
+    //   this.icon.start();
+    //   this.transferIcon = new BluetoothTransferIcon(this);
+    //   this.transferIcon.start();
+    //   this.headphoneIcon = new BluetoothHeadphoneIcon(this);
+    //   this.headphoneIcon.start();
+    // }.bind(this)).catch(function(err) {
+    //   console.error(err);
+    // });
+  },
+
+  /**
+   * Remove all EventListeners and adapters when default adapter
+   * is changed.
+   */
+  _removeEventListeners() {
+    this._bluetooth.removeEventListener('attributechanged',
+      this._defaultAdapterChangeHandler);
+    this._adapter.removeEventListener('attributechanged',
+      this._stateChangeHandler);
+    // remove getAdapter cached promise
+    if (this._cacheGetAdapterPromise) {
+      this._bluetooth.removeEventListener('attributechanged',
+        this._promiseAdapterChangeHandler);
+      this._cacheGetAdapterPromise = null;
+    }
+    this._adapter = null;
+  },
+
+  /**
+   * update settings value and enable bluetooth hardware
+   *
+   * @private
+   */
+  _enableHandler: function bt__enableHandler() {
+    this._debug('enabling bluetooth');
+    this.getAdapter().then((adapter) => {
+      adapter.enable().then(() => { //resolve
+        this._debug('bluetooth enabled');
+        this._updateProfileState(adapter);
+      }, () => { //reject
+        this._debug('can not get bluetooth adapter');
+      });
+    });
+  },
+
+  /**
+   * update settings value and disable bluetooth hardware
+   *
+   * @private
+   */
+  _disableHandler: function bt__disableHandler() {
+    this._debug('disabling bluetooth');
+    this.getAdapter().then((adapter) => {
+      adapter.disable().then(() => { //resolve
+        this._debug('bluetooth disabled');
+      }, () => { //reject
+        this._debug('can not get bluetooth adapter');
+      });
+    });
+  },
+
+  /**
+   * Maintain connect stat of supported profiles.
+   *
+   * @private
+   * @param  {Object} adapter bluetooth adapter
+   */
+  _updateProfileState: function bt__updateProfileState(adapter) {
```

Comment:
The function name is not fit that you have done in following line. These are add event listener for status of each connection profile. Please rename the function name here.